manifests.go

```diff
@@ -0,0 +1,66 @@
+package distribution
+
+import (
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/reference"
+)
+
+// Manifest represents a registry object specifying a target and a set of constituents
+type Manifest interface {
+	// Target returns a descriptor for the configuration of this manifest.  This
+	// may return a nil descriptor if none exists for this manifest.
+	Target() Descriptor
+
+	// Constituents returns a list of objects which make up this manifest.
+	// The dependencies are strictly ordered from base to head. A constituent
+	// is anything which can be represented by a distribution.Descriptor
+	Constituents() []Descriptor
```

Comment:
I wasn't sure `Dependencies` really captured the right meaning as it implied to me something external to this object.  Constituents is more accurate, but I'm not thrilled with the word (I also considered `objects` and `references`).  I was hoping for a minor bikeshed over this one: @dmcgowan @aaronlehmann